src/models/StyledNativeComponent.js

```diff
@@ -17,6 +18,16 @@ type State = {
   generatedStyles: any,
 }
 
+let warnExtendDeprecated
+if (process.env.NODE_ENV !== 'production') {
+  warnExtendDeprecated = once(() => {
+    // eslint-disable-next-line no-console
+    console.error(
+      'Warning: The "extend" API has been removed in the upcoming v4.0 release. Use styled(StyledComponent) instead. You can find more information here: https://github.com/styled-components/styled-components/issues/1546'
```

Comment:
Maybe change the wording to `The "extend" API will be removed... ...due to issues with server-side rendering.`?

Reply:
There were other reasons too. Linking to the issue isn't enough?

Reply:
Oh yeah I didn't see that, but we should still say "will be"?

Reply:
Sure